src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseSign1Message.cs

```diff
@@ -0,0 +1,283 @@
+            return SignCore(encodedProtectedHeader, GetEmptyCborMap(), signature, content, isDetached);
+        }
+
+        [UnsupportedOSPlatform("browser")]
+        public static byte[] Sign(byte[] content, CoseHeaderMap protectedHeaders, CoseHeaderMap unprotectedHeaders, AsymmetricAlgorithm key, bool isDetached = false)
+        {
+            if (content == null)
+            {
+                throw new ArgumentNullException(nameof(content));
+            }
+
+            if (key == null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            if (protectedHeaders == null)
+            {
+                throw new ArgumentNullException(nameof(protectedHeaders));
+            }
+
+            if (unprotectedHeaders == null)
+            {
+                throw new ArgumentNullException(nameof(unprotectedHeaders));
+            }
+
+            KeyType keyType = key switch
+            {
+                ECDsa => KeyType.ECDsa,
+                RSA => KeyType.RSA,
+                _ => throw new CryptographicException(SR.Format(SR.Sign1SignUnsupportedKey, key.GetType()))
+            };
+
+            ThrowIfDuplicateLabels(protectedHeaders, unprotectedHeaders);
+
+            ReadOnlyMemory<byte> encodedAlg = GetCoseAlgorithmFromBuckets(protectedHeaders, unprotectedHeaders);
+
+            int? algorithmHeader = DecodeCoseAlgorithmHeader(encodedAlg);
+            Debug.Assert(algorithmHeader.HasValue, "Algorithm (alg) is a known header and should have been validated in Set[Encoded]Value()");
+
+            HashAlgorithmName hashAlgorithm = GetHashAlgorithmFromCoseAlgorithmAndKeyType(algorithmHeader.Value, keyType);
+
+            byte[] encodedProtetedHeaders = protectedHeaders.Encode(mustReturnEmptyBstrIfEmpty: true);
+            byte[] toBeSigned = CreateToBeSigned(SigStructureCoxtextSign1, encodedProtetedHeaders, content);
+
+            byte[] signature = keyType == KeyType.ECDsa ?
+                SignWithECDsa((ECDsa)key, toBeSigned, hashAlgorithm) :
+                SignWithRSA((RSA)key, toBeSigned, hashAlgorithm);
+
+            return SignCore(encodedProtetedHeaders, unprotectedHeaders.Encode(), signature, content, isDetached);
+        }
+
+        // Validate duplicate labels https://datatracker.ietf.org/doc/html/rfc8152#section-3.
+        internal static void ThrowIfDuplicateLabels(CoseHeaderMap @protected, CoseHeaderMap unprotected)
+        {
+            foreach ((CoseHeaderLabel Label, ReadOnlyMemory<byte>) header in @protected)
+            {
+                if (unprotected.TryGetEncodedValue(header.Label, out _))
+                {
+                    throw new CryptographicException(SR.Sign1SignHeaderDuplicateLabels);
+                }
+            }
+        }
+
+        private static byte[] SignCore(ReadOnlySpan<byte> encodedProtectedHeader, ReadOnlySpan<byte> encodedUnprotectedHeader, ReadOnlySpan<byte> signature, ReadOnlySpan<byte> content, bool isDetached)
```

Comment:
Some of these lines are fairly long, and this one has parameters that start at a position later than GitHub's default width... so it's definitely too long :smile:.

```suggestion
        private static byte[] SignCore(
            ReadOnlySpan<byte> encodedProtectedHeader,
            ReadOnlySpan<byte> encodedUnprotectedHeader,
            ReadOnlySpan<byte> signature,
            ReadOnlySpan<byte> content,
            bool isDetached)
```